src/sentry/tasks/relocation.py

```diff
@@ -131,9 +145,197 @@
+    retry_backoff=RETRY_BACKOFF,
+    retry_backoff_jitter=True,
+    soft_time_limit=FAST_TIME_LIMIT,
+)
+def uploading_start(uuid: str, replying_region_name: str | None, org_slug: str | None) -> None:
+    """
+    The very first action in the relocation pipeline. In the case of a `SAAS_TO_SAAS` relocation, it
+    will trigger the export of the requested organization from the region it currently live in. If
+    this is a `SELF_HOSTED` relocation, this task is a no-op that merely auto-triggers the next step
+    in the chain, `upload_complete`.
+
+    In the case of a `SAAS_TO_SAAS` relocation, we'll need to export an organization from the
+    exporting region (ER) back to the requesting region (RR - where this method is running). Because
+    region-to-region messaging is forbidden, all of this messaging will need to be proxied via the
+    control silo (CS). Thus, to accomplish this export-and-copy operation, we'll need to use
+    sequenced RPC calls to fault-tolerantly execute code in these three siloed locations.
+
+    Due to of our system architecture, the sequence of remote functions executed can be a bit hard
+    to follow, so it is diagrammed and listed below. Each function executed is given a sequential
+    numerical identifier and is annotated with information about where to find the source and which
+    silo it will be executed in:
+
+
+        | Requesting |            |   Control  |            | Exporting  |
+        |    (RR)    |            |    (CS)    |            |    (ER)    |
+        |============|            |============|            |============|
+        |            |            |            |            |            |
+        |     01     |            |            |            |            |
+        |            |-----02---->|            |            |            |
+        |            |            |     03     |            |            |
+        |            |            |     04     |            |            |
+        |            |            |            |-----05---->|            |
+        |            |            |            |            |     06     |
+        |            |            |            |            |     07     |
+        |            |            |            |<----08-----|            |
+        |            |            |     09     |            |            |
+        |            |            |     10     |            |            |
+        |            |<----11-----|            |            |            |
+        |     12     |            |            |            |            |
+        |            |            |            |            |            |
+
+
+    01. (RR) .../tasks/relocation.py::uploading_start: This first function grabs this (aka the
+        "requesting" region) region's public key, then sends an RPC call to the control silo,
+        requesting an export of some `org_slug` from the `replying_region_name` in which is lives.
```

Comment:
Is the `replying_region_name` the same as the exporting region?

Reply:
Yes. The only reason I went with `exporting region` here is because `replying region` would have the same acronym. I can change if it is too confusing.